packages/zoe/src/contracts/multipoolAutoswap.js

```diff
@@ -0,0 +1,627 @@
+        rejectIfNotTokenBrand(inviteHandle, brandOut);
+
+        const {
+          poolHandle: poolHandleA,
+          newUserAmounts: newUserAmountsA,
+          newPoolAmounts: newPoolAmountsA,
+        } = performSwap(
+          harden({
+            userAllocation: zoe.getCurrentAllocation(
+              inviteHandle,
+              harden([keywordIn, CENTRAL_TOKEN]),
+            ),
+            keywordIn,
+            keywordOut: CENTRAL_TOKEN,
+            secondaryBrand: brandIn,
+          }),
+        );
+        const {
+          poolHandle: poolHandleB,
+          newUserAmounts,
+          newPoolAmounts: newPoolAmountsB,
+        } = performSwap(
+          harden({
+            userAllocation: newUserAmountsA,
+            keywordIn: CENTRAL_TOKEN,
+            keywordOut,
+            secondaryBrand: brandOut,
+          }),
+        );
+        const keywords = harden([keywordIn, keywordOut, CENTRAL_TOKEN]);
+        const amountMaths = zoe.getAmountMaths(keywords);
+        const finalPoolAmountsA = {
+          ...newPoolAmountsA,
+          [keywordOut]: amountMaths[keywordOut].getEmpty(),
+        };
+        const finalPoolAmountsB = {
+          ...newPoolAmountsB,
+          [keywordIn]: amountMaths[keywordIn].getEmpty(),
+        };
+        const finalUserAmounts = {
+          ...newUserAmounts,
+          [keywordIn]: newUserAmountsA[keywordIn],
+        };
+        zoe.reallocate(
+          harden([poolHandleA, poolHandleB, inviteHandle]),
+          harden([finalPoolAmountsA, finalPoolAmountsB, finalUserAmounts]),
+          keywords,
+        );
+        zoe.complete(harden([inviteHandle]));
+        return `Swap successfully completed.`;
+      },
+    });
+    const { invite, inviteHandle } = zoe.makeInvite(seat, {
+      seatDesc: 'autoswapSeat',
+    });
+    return invite;
+  };
+
+  return harden({
+    invite: makeInvite(),
+    publicAPI: {
+      getBrandKeywordRecord: () => {
+        const { issuerKeywordRecord } = zoe.getInstanceRecord();
+        const brandKeywordRecord = {};
+        Object.entries(issuerKeywordRecord).forEach(([keyword, issuer]) => {
+          const { brand } = zoe.getIssuerRecord(issuer);
+          brandKeywordRecord[keyword] = brand;
+        });
+        return harden(brandKeywordRecord);
+      },
+      getKeywordForBrand: brand => {
+        assert(
+          liquidityTable.has(brand),
+          details`There is no pool for this brand. To create a pool, call 'addPool'`,
+        );
+        return liquidityTable.get(brand).tokenKeyword;
+      },
+      addPool,
+      getPoolAllocation,
+      getLiquidityIssuer: tokenBrand =>
+        liquidityTable.get(tokenBrand).liquidityIssuer,
+      makeInvite,
+      /**
+       * `getPrice` calculates the result of a trade, given a certain
+       * amount of digital assets in.
+       * @param {object} amountIn - the amount of digital assets to be
+       * sent in
+       */
+      getPrice: (amountIn, brandOut) => {
```

Comment:
should this be `getCurrentPrice()`? It's just an instantaneous quote, with no hint that the price will remain stable.

Reply:
Sure, that sounds good. I'll change autoswap too.